tests/framework/unittest/dynamic_hocon_unit_tests.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from typing import List
+
+from unittest import TestCase
+
+from neuro_san.internals.utils.file_of_class import FileOfClass
+
+from tests.framework.driver.data_driven_agent_test_driver import DataDrivenAgentTestDriver
+from tests.framework.unittest.unit_test_assert_forwarder import UnitTestAssertForwarder
+
+
+class DynamicHoconUnitTests:
```

Comment:
Class which assists with creating parameterized unit test methods out of individual hocon file references.
Hopefully the class comment below begins to explain everything.
It's pretty abstract, so see test_hocons.py for an example as to how this gets used in a concrete fashion.
The goal here is that this guy absorbs the complexity so that the test_hocons.py of the world can stay simple.